### PR TITLE
SD35Large T3K Model Bring up

### PR DESCRIPTION
**Pull request by mbahnasTT**
### Ticket
SD35Large T3K Model Bring up

### What's changed
- Model SD3.5
- Allgather matmul OP
- LayerNorm Distributed OP

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15216602400) CI passes
- [x] [T3K Nightly tests](https://github.com/tenstorrent/tt-metal/actions/runs/15216643528) CI passes
- [x] [T3K Frequent tests](https://github.com/tenstorrent/tt-metal/actions/runs/15216532608) CI passes
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [x] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/demos/yolov4/tests/perf/test_perf.py [hunk 1]
```diff
@@ -15,6 +15,13 @@
 from models.utility_functions import disable_persistent_kernel_cache, profiler
 
 
+def get_expected_compile_time_sec():
+    return 75
+
+
+def get_expected_inference_time_sec():
+    return 0.37
+
```
Comment (rdraskicTT) [inline models/demos/yolov4/tests/perf/test_perf.py:24]: Looks like this is not used anywhere. These are parameterized in the fixture below, could you please remove the methods and update the fixture if necessary?

File: models/experimental/stable_diffusion_35_large/demo.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    "mesh_device",
+    [
+        {"N150": (1, 1), "N300": (1, 2), "T3K": (1, 8), "TG": (8, 4)}.get(
+            os.environ.get("MESH_DEVICE"), len(ttnn.get_device_ids())
+        )
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "model_name, image_w, image_h, guidance_scale, num_inference_steps",  # "prompt_sequence_length", "spatial_sequence_length",
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/demo.py:28]: Why are the last two parameters commented out? Should they be removed?

File: models/experimental/stable_diffusion_35_large/demo.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    [
+        #        ("medium", 512, 512, 4.5, 40, 333, 1024),
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/demo.py:30]: Why are three lines commented out?

File: models/experimental/stable_diffusion_35_large/tests/test_patch_embedding.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        torch_input_tensor.permute([0, 2, 3, 1]),  # BCYX -> BYXC
+        device=mesh_device,
+        mesh_mapper=ttnn.ReplicateTensorToMesh(mesh_device),
+        layout=ttnn.TILE_LAYOUT,
+        dtype=dtype,
+    )
+
+    tt_output = tt_model(tt_input_tensor)
+
+    assert_quality(torch_output, tt_output, pcc=0.999_990, shard_dim=-1)
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/tests/test_patch_embedding.py:91]: Why is there a GitHub Actions failure comment?

File: models/experimental/stable_diffusion_35_large/tests/test_patch_embedding_conv2d.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        # (1, 128, 128, (3, 3), (1, 1), 1024, 1024),
+        # (1, 128, 3, (3, 3), (1, 1), 1024, 1024),
+        # (1, 16, 512, (3, 3), (1, 1), 128, 128),
+        # (1, 256, 128, (3, 3), (1, 1), 1024, 1024),
+        # (1, 256, 256, (3, 3), (1, 1), 1024, 1024),
+        # (1, 256, 256, (3, 3), (1, 1), 512, 512),
+        # (1, 512, 512, (3, 3), (1, 1), 128, 128),
+        # (1, 512, 512, (3, 3), (1, 1), 256, 256),
+        # (1, 512, 256, (3, 3), (1, 1), 512, 512),
+        # (1, 512, 512, (3, 3), (1, 1), 512, 512),
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/tests/test_patch_embedding_conv2d.py:46]: Why are these commented out?

File: models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        spatial_padded_4d, dtype=ttnn_dtype, device=mesh_device, layout=ttnn.TILE_LAYOUT, shard_dim=-1
+    )
+
+    ##
+    prompt_padded_4d = prompt.unsqueeze(1)
+    if pad_embedding_dim:
+        prompt_padded_4d = torch.nn.functional.pad(
+            prompt_padded_4d, pad=(0, hidden_dim_padding), mode="constant", value=0
+        )
+    # tt_prompt =from_torch_fast(prompt_padded_4D, dtype=ttnn_dtype, device=mesh_device, layout=ttnn.TILE_LAYOUT, shard_dim=None)
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py:114]: Remove?

File: models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        )
+
+    with torch.no_grad():
+        spatial_output, prompt_output = torch_model(spatial=spatial, prompt=prompt, time_embed=time)
+
+    tt_spatial_output_padded, tt_prompt_output_padded = tt_model(
+        spatial=tt_spatial, prompt=tt_prompt, time_embed=tt_time, N=spatial_sequence_length, L=prompt_sequence_length
+    )
+
+    # num_measurement_iterations=38
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py:137]: Please explain why this is here as part of the comment or remove.

File: ttnn/cpp/ttnn/operations/normalization/layernorm_distributed/device/multi_core/layernorm_post_all_gather_op_multi_core.cpp [hunk 8]
```diff
@@ -141,7 +141,7 @@ tt::tt_metal::operation::ProgramWithCallbacks layernorm_post_allgather_multi_cor
     auto a_addr = a.buffer()->address();
     auto stats_addr = stats.buffer()->address();
     auto gamma_dram_addr = gamma.has_value() ? gamma.value().buffer()->address() : 0;
-    TT_FATAL(gamma_dram_addr != 0, "Gamma must be provided");
+    // TT_FATAL(gamma_dram_addr != 0, "Gamma must be provided");
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/layernorm_distributed/device/multi_core/layernorm_post_all_gather_op_multi_core.cpp:144]: Should be removed?

File: models/experimental/stable_diffusion_35_large/tests/test_fun_transformer.py [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    #     )
+    # profiler.end(f"run")
+    # devices = mesh_device.get_devices()
+    # ttnn.DumpDeviceProfiler(devices[0])
+    # total_time = profiler.get("run")
+    # avg_time = total_time / num_measurement_iterations
+    # print(f" TOTAL TIME: {total_time} AVG TIME: {avg_time}\n")
+
+    torch_output = torch.unsqueeze(torch_output, 1)
+    print(f"tt_output shape {tt_output.shape} torch_output {torch_output.shape}")
```
Comment (bbradelTT) [inline models/experimental/stable_diffusion_35_large/tests/test_fun_transformer.py:165]: Should use logger?

**Review by bbradelTT — APPROVED** [review]

File: models/experimental/stable_diffusion_35_large/demo.py (hunk 3 above)
Comment (mbahnasTT) [reply models/experimental/stable_diffusion_35_large/demo.py:30]: These are the different variants of the model. We'll enable them later.

File: models/experimental/stable_diffusion_35_large/demo.py (hunk 2 above)
Comment (mbahnasTT) [reply models/experimental/stable_diffusion_35_large/demo.py:28]: OK, I'll remove it.

File: models/experimental/stable_diffusion_35_large/demo.py (hunk 3 above)
Comment (bbradelTT) [reply models/experimental/stable_diffusion_35_large/demo.py:30]: Please add a comment to that effect.
Commented out code should either:
- be removed (preferred) or
- have an explanation about why it is there.

File: models/experimental/stable_diffusion_35_large/tests/test_patch_embedding.py (hunk 4 above)
Comment (mbahnasTT) [reply models/experimental/stable_diffusion_35_large/tests/test_patch_embedding.py:95]: I fixed it now. Thanks!

File: models/experimental/stable_diffusion_35_large/tests/test_patch_embedding_conv2d.py (hunk 5 above)
Comment (mbahnasTT) [reply models/experimental/stable_diffusion_35_large/tests/test_patch_embedding_conv2d.py:46]: These are for the other variants of the model (i.e. model size and resolutions). We'll enable them, when moving the model from experimental to demos.

File: models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py (hunk 7 above)
Comment (mbahnasTT) [reply models/experimental/stable_diffusion_35_large/tests/test_transformer_block.py:138]: removed

**Review by ayerofieiev-tt — COMMENTED** [review]
Lets extract these changes + their tests into separate PRs

- Allgather matmul OP
- LayerNorm Distributed OP
